File: pastebin/clipboard.cpp
#include <Windows.h>
#include <stdio.h>

BOOL CopyToClipboard(char* pszData)
{
    if(::OpenClipboard(NULL))
    {
        ::EmptyClipboard();
        HGLOBAL clipbuffer;
        char *buffer;
        clipbuffer = ::GlobalAlloc(GMEM_DDESHARE, strlen(pszData)+1);
        buffer = (char *)::GlobalLock(clipbuffer);
        strcpy_s(buffer,strlen(pszData)+1, pszData);
        ::GlobalUnlock(clipbuffer);
        ::SetClipboardData(CF_TEXT, clipbuffer);
        ::CloseClipboard();
        return TRUE;
    }
    return FALSE;
}

BOOL GetTextFromClipboard()
{
    if(::OpenClipboard(NULL))
    {
        
        HGLOBAL hMem = GetClipboardData(CF_TEXT);
        if(NULL != hMem)
        {
            char* lpStr = (char*)::GlobalLock(hMem); 
            if(NULL != lpStr)
            {
                printf("%s",lpStr);
                ::GlobalUnlock(hMem);
            }
        }
        ::CloseClipboard();
        return TRUE;
    }
    return FALSE;
}

File: pastebin/clipboard.py
import win32clipboard
import win32con

def getTextFromClipboard():

    win32clipboard.OpenClipboard(0)
    data = win32clipboard.GetClipboardData(win32con.CF_TEXT)
    
    # TODO: deal with different file format
    print(data.decode("utf-8"))
    win32clipboard.CloseClipboard()

def copyToClipboard(buf:str):
    win32clipboard.OpenClipboard()
    win32clipboard.EmptyClipboard()
    win32clipboard.SetClipboardText(buf)
    win32clipboard.CloseClipboard()

if __name__ == '__main__':
    getTextFromClipboard()
    copyToClipboard("Hello World")
    getTextFromClipboard()